ui/Screen/Identity/IdentityCreationForm.svelte

```diff
@@ -0,0 +1,189 @@
+<script>
+  import { gql } from "apollo-boost";
+  import { getClient, mutate } from "svelte-apollo";
+  import validatejs from "validate.js";
+
+  import {
+    avatarUrlStore,
+    displayNameStore,
+    handleStore,
+    shareableEntityIdentifierStore
+  } from "../../store/identity.js";
+
+  import { Button, Input, Text, Title } from "../../DesignSystem/Primitive";
+
+  export let onSuccess, onError;
+
+  const HANDLE_MATCH = "^[a-z0-9][a-z0-9_-]+$";
+  const DISPLAY_NAME_MATCH = "^[a-z0-9 ]+$";
+
+  let handle,
+    displayName,
+    avatarUrl,
+    validations,
+    beginValidation = false;
+
+  validatejs.options = {
+    fullMessages: false
+  };
+
+  validatejs.validators.optional = (value, options) => {
+    return !validatejs.isEmpty(value)
+      ? validatejs.single(value, options)
+      : null;
+  };
+
+  const constraints = {
+    handle: {
+      presence: {
+        message: "You must provide a handle",
+        allowEmpty: false
+      },
+      format: {
+        pattern: new RegExp(HANDLE_MATCH, "i"),
+        message: `Handle should match ${HANDLE_MATCH}`
+      }
+    },
+    displayName: {
+      optional: {
+        format: {
+          pattern: new RegExp(DISPLAY_NAME_MATCH, "i"),
+          message: `Display name should match ${DISPLAY_NAME_MATCH}`
+        }
+      }
+    },
+    avatarUrl: {
+      optional: {
+        url: {
+          schemes: ["http", "https"],
+          message: "Not a valid image URL",
+          allowLocal: false
+        }
+      }
+    }
+  };
+
+  const validate = () => {
+    if (!beginValidation) {
+      return;
+    }
+    validations = validatejs(
+      {
+        handle: handle,
+        displayName: displayName,
+        avatarUrl: avatarUrl
+      },
+      constraints
+    );
+  };
+
+  $: validate(handle, displayName, avatarUrl);
+
+  const client = getClient();
+
+  const CREATE_IDENTITY = gql`
+    mutation($handle: String!, $displayName: String, $avatarUrl: String) {
+      createIdentity(
+        handle: $handle
+        displayName: $displayName
+        avatarUrl: $avatarUrl
+      ) {
+        id
+        shareableEntityIdentifier
+        metadata {
+          handle
+          displayName
+          avatarUrl
```

Comment:
Can you also request the `avatarFallback` here, so it can be filled in the store as well?

Reply:
Done! This is a great example of something we can throw in a data access layer at some point & make our lives a lot easier 😇

Reply:
I'm not sure I follow @sarahscott, did you forget to do the change? I still see that you're doing a separate query in `IdentityCreationSuccess.svelte` to retrieve the `avatarFallback`. It would be better to fetch it right here where you create the identity.

I initially also overlooked the `fallbackAvatar`, here's what the query could look like:
<img width="1062" alt="Screenshot 2020-03-23 at 18 09 41" src="https://user-images.githubusercontent.com/158411/77343414-ecaff100-6d31-11ea-9f94-3f9b9ead8d06.png">